internal/controller/preflightchecks/preflight_checks.go

```diff
@@ -238,3 +252,26 @@ func isPredefinedProvider(providerName string, providerType clusterctlv1.Provide
 
 	return err == nil, nil
 }
+
+// versionChanged try to get installed version from provider status and decide if it has changed.
+func versionChanged(provider genericprovider.GenericProvider) (bool, error) {
+	installedVersion := provider.GetStatus().InstalledVersion
+	if installedVersion == nil {
+		return true, nil
+	}
+
+	currentVersion, err := version.ParseSemantic(*installedVersion)
+	if err != nil {
+		return false, err
+	}
+
+	res, err := currentVersion.Compare(provider.GetSpec().Version)
+	if err != nil {
+		return false, err
+	}
+
+	// we need to delete installed components if versions are different
+	versionChanged := res != 0
+
+	return versionChanged, nil
+}
```

Comment:
I know this is just codemotion, but maybe we could make this even more useful and if
`currentVersion.Compare != 0` perform a [deep.Equal()](https://pkg.go.dev/github.com/go-test/deep#Equal) and return the `[]string` with the diff between the two or else return `[]string{}, nil`.
This way we could log what is the detected difference and provide useful info to the user about it.